kafka-client/src/main/java/dev/responsive/kafka/store/CommitBuffer.java

```diff
@@ -63,14 +63,16 @@ class CommitBuffer<K> implements RecordBatchingStateRestoreCallback {
   private final Supplier recordCollector;
   private final TopicPartition changelog;
   private final BufferPlugin<K> plugin;
+  private final boolean truncateChangelog;
 
   CommitBuffer(
       final CassandraClient client,
       final String tableName,
       final TopicPartition changelog,
       final RecordCollector.Supplier recordCollector,
       final Admin admin,
-      final BufferPlugin<K> plugin
+      final BufferPlugin<K> plugin,
+      final boolean truncateChangelog
```

Comment:
the UX of moving from compact -> delete & truncate would be to first change configs, then restart the node. I think that makes sense to me - I do wonder if we want a per-topology config to turn off truncation entirely, but that can be a follow-up.

Reply:
I'll let @ableegoldman chime in here. As per our discussion yesterday she was planning on experimenting with some options here.

Reply:
Yeah it's not worth worrying about this until we settle on the broader API, but fwiw I agree -- we should expose this with both an application/topology-wide default* and a store-level granular optimization. I'd imagine most users either want it on for everything or nothing. Then a handful who will possibly want it on as the topology default, but with carveouts for select changelogs that are eg read downstream or by another app